Index: ext/standard/basic_functions.c
===================================================================
RCS file: /repository/php-src/ext/standard/basic_functions.c,v
retrieving revision 1.725.2.31.2.68
diff -u -r1.725.2.31.2.68 basic_functions.c
--- ext/standard/basic_functions.c	31 Dec 2007 07:20:12 -0000	1.725.2.31.2.68
+++ ext/standard/basic_functions.c	29 Mar 2008 21:17:12 -0000
@@ -3105,8 +3105,13 @@
 zend_function_entry basic_functions[] = {
 	PHP_FE(constant,														arginfo_constant)
 	PHP_FE(bin2hex,															arginfo_bin2hex)
-	PHP_FE(sleep,															arginfo_sleep)
-	PHP_FE(usleep,															arginfo_usleep)
+#if GNUPTH
+	ZEND_RAW_FENTRY("sleep", ZEND_FN(pth_sleep),	arginfo_sleep, 0)
+	ZEND_RAW_FENTRY("usleep", ZEND_FN(pth_usleep),	arginfo_usleep, 0)
+#else
+	PHP_FE(sleep,								arginfo_sleep)
+	PHP_FE(usleep,								arginfo_usleep)
+#endif
 #if HAVE_NANOSLEEP
 	PHP_FE(time_nanosleep,													arginfo_time_nanosleep)
 	PHP_FE(time_sleep_until,												arginfo_time_sleep_until)
